### PR TITLE
Updated README.md

### LINKED ISSUE #55: Adding Live Contributor Profile Photos to the README.md
## Issue Title
Adding Live Contributor Profile Photos to the README.md

## Type

- [ ] Big
- [ ] Feature Request
- [ ] Info
- [ ] Bug
- [x] Documentation
- [ ] Other (please specify):

## Description
Enhance community engagement and personalize our project's README file by including live contributor profile photos. This addition will foster a sense of community, improve branding, and recognize contributors.

Currently, the README file lacks visual representation of contributors, which diminishes community engagement and personal connection.

## Steps to Reproduce

1. Open the project's README.md file.
2. Observe the lack of contributor photos.
3. Note the absence of personalized engagement features.

## Screenshots / Code Snippets (if applicable)
Current : - 
Screenshot (43)

Example, after the issue solved: - 
Screenshot (44)

## Environment

- Operating System: Not applicable 
- Browser (if applicable): Not applicable
- Version/Commit ID (if applicable): Not applicable

## Possible Solution (if you have any in mind)
Integrate live contributor profile photos using Markdown or HTML elements that dynamically source images from GitHub profiles. This can be implemented using a simple script to fetch and display the images.

## Additional Information
Implementing this feature will not only beautify our README but also strengthen our project's collaborative spirit, making contributions visually impactful.

### PR DESCRIPTION
**Pull request by jaidh01**
## Description

This update revamps the README.md file for the FaceRec project to improve clarity and structure. The changes enhance user engagement and address issue #55  related to documentation quality.

Fixes #55 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Changes were verified by reviewing the rendered output of the updated README.md on GitHub. Tests included:

- [x] Checked formatting and links.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,176 +19,103 @@
...
-1. `create_new_faceEntry()`: This function receives a POST request with an image and metadata. It extracts the face from the image, calculates the embeddings, and stores the data in the database.
-
-2. `Data()`: This function sent a GET request  to `/data` endpoint of FastAPI app to get the list of Face Entries from MongoDB.
-
-3. `update()` :This function is used to update the details of the face entry in the database.
-
-4. `read()` : This function sent a GET request with specific Employeecode to read the information related to that particular Employeecode.
-
-5. `delete()` : This function is used to delete the specific Employee Data.
-
```
Comment (Devasy23) [inline README.md:179]: This is required, maybe you can add sequence diagram to picture it more efficiently and in an appealing was

File: README.md [hunk 2]
```diff
@@ -15,176 +19,103 @@
...
-The database contains a `faceEntries` collection with the following schema:
-
-- `id`: A unique identifier for the face entry.
-- `Employeecode`: A unique  employee ID associated with the image.
-- `Name`: The name of the person in the image.
-- `gender`: The gender of the person.
-- `Department`: The department of the person
-- `time`: The time the face entry was created.
-- `embeddings`: The embeddings of the face image.
-- `Images`: Base64 encoded image file.
```
Comment (Devasy23) [inline README.md:166]: This is also required but not in the readme.md but in the sphinx docs so can you add that as well

**Comment by Devasy23** [comment]
@jaidh01 the pre-commit hook messed with the requirement.txt maybe that's why the pipeline isn't passing.


`DEBUG` that issue and if you can revamp some changes to the sphinx doc as well then I'll assign this ticket, level2 tag

**Comment by Devasy23** [comment]
Everything else looks good to me, wbu @devansh-shah-11 ?!

**Comment by jaidh01** [comment]
Hi @Devasy23, I have done all the changes as you suggested.
Please spare a time to review it.

File: README.md (hunk 1 above)
Comment (devansh-shah-11) [reply README.md]: @jaidh01 do make a sequence diagram

**Comment by Devasy23** [comment]
You still need to add the sequence diagram

**Comment by jaidh01** [comment]
> You still need to add the sequence diagram

I have already added it.
Screenshot (62)

File: README.md [hunk 3]
```diff
@@ -15,176 +19,156 @@
...
-To start FLask and FastAPI, run the given command:
-```bash
-python main.py
+1. `create_new_faceEntry()`: Receives a POST request with an image and metadata. It extracts the face, calculates the embeddings, and stores the data in the database.
+2. `Data()`: Sends a GET request to the `/data` endpoint of the FastAPI app to get the list of Face Entries from MongoDB.
+3. `update()`: Updates the details of the face entry in the database.
+4. `read()`: Sends a GET request with a specific `Employeecode` to read the related information.
+5. `delete()`: Deletes the specific employee data.
+
+## Sequence Diagram
```
Comment (jaidh01) [inline README.md:103]: From here the sequence diagram starts.

Comment (Devasy23) [reply README.md:81]: This is the plant uml for diagram u may need to compile and render it and then put it here

**Comment by Devasy23** [comment]
And are you looking forward to add this changes to the sphinx documentation also, that is in the .rst files from `docs/`

**Comment by jaidh01** [comment]
@Devasy23, could you please guide me through updating the Sphinx documentation? I have no prior experience with this, but I am eager to learn.

**Comment by Devasy23** [comment]
Hi @jaidh01 ,

Thank you for updating the README.md. To ensure consistency across our documentation, I'd like those changes to be reflected in our Sphinx documentation as well.

Sphinx is a powerful tool for generating documentation, typically used for Python projects. It processes .rst (reStructuredText) files, but you can also include Markdown files like README.md with a few configurations. Here’s a quick overview of what you'll need to do:

1. Install necessary extensions:

Sphinx doesn't support Markdown out of the box, but you can add support for it with extensions. Please install recommonmark and myst-parser, two popular extensions for parsing Markdown files:

pip install recommonmark myst-parser

2. Update conf.py:

In the Sphinx configuration file (conf.py), you’ll need to enable these extensions. Add or update the following lines in the file:

extensions = ['recommonmark', 'myst_parser']

# To automatically include the README.md, add this:
source_suffix = {
    '.rst': 'restructuredtext',
    '.md': 'markdown',
}

3. Include README.md in your index.rst:

In the index.rst (or wherever appropriate in your documentation), include the README.md file like this:

.. include:: ../README.md
   :parser: myst

This will ensure that the changes from the README.md are parsed and included in the Sphinx-generated documentation.

References:

Sphinx official documentation: https://www.sphinx-doc.org/

Guide on using Markdown with Sphinx: https://myst-parser.readthedocs.io/

recommonmark GitHub repo: https://github.com/readthedocs/recommonmark


Let me know if you need any further clarification!

**Comment by Devasy23** [comment]
Any update on this @jaidh01 ?

**Comment by Devasy23** [comment]
Any update on this @jaidh01 ?

**Comment by jaidh01** [comment]
> Any update on this @jaidh01 ?

I will update the sphinx docs by today evening.

**Comment by jaidh01** [comment]
Hi @Devasy23, I have updated the sphinx docs.
And in the code, there was only single line change in the `index.rst` file. Where I just add this line: - `:parser: myst` at the end.

So, should I commit it or leave it?

**Comment by Devasy23** [comment]
@jaidh01 please take pull from main branch to solve conflicts

**Comment by jaidh01** [comment]
> @jaidh01 please take pull from main branch to solve conflicts

I have taken the pull from the main branch, and I think it should now resolve the conflicts.

**Comment by Devasy23** [comment]
> > @jaidh01 please take pull from main branch to solve conflicts
> 
> I have taken the pull from the main branch, and I think it should now resolve the conflicts.

Cool, 
- I saw you sphinx changes but there aren't any, the readthe docs isn;t still parsing README.md file.
image

- You haven't put seq diagram, just the plantuml code that LLM generated.
image

**Pull request merged**